### PR TITLE
sql: fix create statistic re-parse issues

### PR DESCRIPTION
**Pull request by e-mbrown**
The CREATE STATISTICS statement had re-parse issues because it
requires a float constant. This change replaces the underscore with a
float value between 1.0 and 0.0 when using the `FmtHideConstants` flag.

Release note: None

### CODE REVIEW

File: pkg/sql/parser/testdata/create_misc [hunk 1]
```diff
@@ -95,8 +94,7 @@ CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING 0.1 AS OF SYSTEM TIME
 ----
 CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING 0.1 AS OF SYSTEM TIME '2016-01-01'
 CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING 0.1 AS OF SYSTEM TIME ('2016-01-01') -- fully parenthesized
-CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING _ AS OF SYSTEM TIME _ -- literals removed
-REPARSE WITHOUT LITERALS FAILS: at or near "_": syntax error
+CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING 0.001 AS OF SYSTEM TIME '_' -- literals removed
```
Comment (rafiss) [inline pkg/sql/parser/testdata/create_misc:97]: is this the output you get?

when i do the rewrite i get
```
CREATE STATISTICS a ON col1 FROM t WITH OPTIONS THROTTLING 0.001 AS OF SYSTEM TIME _ -- literals removed
```

File: pkg/sql/sem/tree/create.go [hunk 2]
```diff
@@ -1917,7 +1917,10 @@ func (o *CreateStatsOptions) Format(ctx *FmtCtx) {
 		// TODO(knz): Remove all this with ctx.FormatNode()
 		// if/when throttling supports full expressions.
 		if ctx.flags.HasFlags(FmtHideConstants) {
-			ctx.WriteByte('_')
+			// Using the value '0.001' instead of '0.0', because
+			// when using '0.0' the statement reparses with missing
```
Comment (rafiss) [inline pkg/sql/sem/tree/create.go:1921]: nit: the comment is not quite accurate. it should say that when using '0.0' the statement does not get formatted with the THROTTLING option.

**Review by rafiss — CHANGES_REQUESTED** [review]

File: pkg/sql/parser/testdata/create_misc (hunk 1 above)
Comment (e-mbrown) [reply pkg/sql/parser/testdata/create_misc]: I get the same, I must have messed up the rebase.

**Review by rafiss — APPROVED** [review]
lgtm!

bors r+

**Pull request merged**